Automattic/wp-calypso
# Framework: Refactor away from `_.identity()`

#### Changes proposed in this Pull Request

Lodash's `identity()` is used many times in our codebase, but it's pretty straightforward and can be replaced with a function that returns its first argument. 

Some of the instances are actually `defaultProps` and can be removed when unnecessary, and some of them are necessary only to satisfy unit tests. For those, we can alternatively provide a `translate` replacement, which is the rest that this PR is doing.

If you want to find out why we're moving away from Lodash, see https://github.com/Automattic/wp-calypso/pull/50368 and p4TIVU-9Bf-p2.

#### Testing instructions

* Verify Calypso builds and smoke test it.
* Verify all tests pass: `yarn run test-client`. Most of the changes are either in tests or covered by tests, so tests passing is a strong indicator here.
* Go through each change one by one and verify it makes sense. 

#### Note

Some pre-existing ESLint errors are expected.
